xtras/src/supervisor.rs

```diff
@@ -144,13 +148,13 @@ where
 impl<T, R, S> Actor<T, R>
 where
     T: xtra::Actor<Stop = S>,
-    R: fmt::Display + Send + 'static,
+    R: Error + Send + Sync + 'static,
     S: Into<R> + Send + 'static,
 {
     pub fn handle(&mut self, msg: Stopped<R>, ctx: &mut Context<Self>) {
         let actor = T::name();
-        let reason_str = msg.reason.to_string();
-        let should_restart = (self.restart_policy)(msg.reason);
+        let should_restart = (self.restart_policy)(&msg.reason);
+        let reason_str = format!("{:#}", anyhow::Error::new(msg.reason)); // Anyhow will format the entire chain of errors when using `alternate` Display (`#`)
```

Comment:
👍 Nice trick. It's interesting that it's not easy to print the error chain without converting to `anyhow::Error`.

Reply:
It is in the works: https://github.com/rust-lang/project-error-handling/issues/10#issuecomment-720652610